src/validation.cpp

```diff
@@ -2745,23 +2744,21 @@ bool CheckBlock(const CBlock& block, CValidationState& state, bool fCheckPOW, bo
     // TODO: Check if this is ok... blockHeight is always the tip or should we look for the prevHash and get the height?
     int blockHeight = chainActive.Height() + 1;
     const Consensus::Params& consensus = Params().GetConsensus();
-    bool fSaplingActive = consensus.NetworkUpgradeActive(blockHeight, Consensus::UPGRADE_V5_0);
     for (const auto& txIn : block.vtx) {
         const CTransaction& tx = *txIn;
         if (!CheckTransaction(
                 tx,
                 fZerocoinActive,
                 state,
                 isBlockBetweenFakeSerialAttackRange(blockHeight),
-                fColdStakingActive,
-                fSaplingActive
+                fColdStakingActive
         ))
             return state.Invalid(false, state.GetRejectCode(), state.GetRejectReason(),
                                              strprintf("Transaction check failed (tx hash %s) %s", tx.GetHash().ToString(), state.GetDebugMessage()));
 
         // No need to check for zerocoin anymore, they are networkely disabled
         // and checkpoints are preventing the chain for any massive reorganization.
-        if (fSaplingActive && tx.ContainsZerocoins()) {
+        if (consensus.NetworkUpgradeActive(blockHeight, Consensus::UPGRADE_V5_0) && tx.ContainsZerocoins()) {
```

Comment:
Better to leave this outside of the block.vtx for loop, so it's calculated only once and not for every tx inside the block.

Reply:
Done.